oauth_device.go

```diff
@@ -29,9 +29,14 @@ func (oa *Flow) DeviceFlow() (*api.AccessToken, error) {
 	if stdout == nil {
 		stdout = os.Stdout
 	}
+
 	host := oa.Host
 	if host == nil {
-		host = GitHubHost("https://" + oa.Hostname)
+		host, err := NewGitHubHost("https://" + oa.Hostname)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing the hostname '%s': %w", host, err)
+		}
+		oa.Host = host
```

Comment:
@mouismail We weren't doing this before, can you explain the benefits of doing this now?

Reply:
I believe the reason behind this was to handle various OAuth hosts, not just GHEC and 
ensures that the `oa.Host` is dynamically assigned if it is not already set.

Reply:
I think this makes sense to me. @jtmcg anything to add?

Reply:
Nope! Makes sense to me, too 👍

Reply:
Ehhh sorry but this change and the one in webapp flow introduce a different panic. You can see by running these tests:

```go
package oauth_test

import (
	"testing"

	"github.com/cli/oauth"
)

func TestDeviceFlowHostname(_ *testing.T) {
	flow := &oauth.Flow{
		Hostname: "github.com",
	}

	_, err := flow.DeviceFlow()
	if err != nil {
		panic(err)
	}
}

func TestWebFlowHostname(_ *testing.T) {
	flow := &oauth.Flow{
		Hostname: "github.com",
	}

	_, err := flow.WebAppFlow()
	if err != nil {
		panic(err)
	}
}
```

The issue is that previously we were assigning `host` with `host =`. However now we're shadowing `host` when we do `host, err :=`. This results in a panic when we try to use `host` and it is nil. I'll create an issue.

Reply:
https://github.com/cli/oauth/issues/69